log/test_output_test.go

```diff
@@ -93,6 +93,27 @@ func TestOutputFailsTestEvenAfterTermination(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// designed for race detector
+func TestOutputSynchronizesHasErrorsAndAppendError(t *testing.T) {
+	m := &fakeTLog{}
+	o := NewTestOutput(m, nopFormatter{})
+	ch := make(chan struct{})
+	go func() {
+		for i := 1; i <= 100; i++ {
+			o.Append("error", "foo")
+		}
+		ch <- struct{}{}
+	}()
+	go func() {
+		for i := 1; i <= 100; i++ {
+			o.HasErrors()
+		}
+		ch <- struct{}{}
+	}()
+	<-ch
+	<-ch
```

Comment:
the test may timeout if an exception is thrown in one of the goroutines, but i'm not sure we will see any meaningful output. did you try `panic` inside one of these for loops to see what it looks like when you fail?

Reply:
I let the race detector fail this test several times. got a solid true negative statistics before introducing solution.